controller/controller_test.go

```diff
@@ -490,6 +510,82 @@ func TestUpdaterIsUpdatedOnK8sUpdates(t *testing.T) {
 				BackendTimeoutSeconds: 20,
 				BackendMaxConnections: defaultMaxConnections,
 			}},
+			defaultConfig(),
+		},
+		{
+			"ingress with default proxy buffer values when not overridden by the ingress definition",
+			createIngressesFixture(ingressHost, ingressSvcName, ingressSvcPort, map[string]string{ingressAllowAnnotation: ""}),
+			createDefaultServices(),
+			[]IngressEntry{{
+				Namespace:             ingressNamespace,
+				Name:                  ingressName,
+				Host:                  ingressHost,
+				Path:                  ingressPath,
+				ServiceAddress:        serviceIP,
+				ServicePort:           ingressSvcPort,
+				LbScheme:              "",
+				Allow:                 []string{},
+				StripPaths:            false,
+				BackendTimeoutSeconds: 10,
+				BackendMaxConnections: defaultMaxConnections,
+				ProxyBufferSize:       2,
+				ProxyBufferBlocks:     3,
+			}},
+			Config{
+				DefaultBackendTimeoutSeconds: backendTimeout,
+				DefaultProxyBufferSize:       2,
+				DefaultProxyBufferBlocks:     3,
+			},
+		},
+		{
+			"ingress definition overrides default proxy buffer values",
+			createIngressesFixture(ingressHost, ingressSvcName, ingressSvcPort, map[string]string{ingressAllowAnnotation: "", proxyBufferSizeAnnotation: "6", proxyBufferBlocksAnnotation: "4"}),
+			createDefaultServices(),
+			[]IngressEntry{{
+				Namespace:             ingressNamespace,
+				Name:                  ingressName,
+				Host:                  ingressHost,
+				Path:                  ingressPath,
+				ServiceAddress:        serviceIP,
+				ServicePort:           ingressSvcPort,
+				LbScheme:              "",
+				Allow:                 []string{},
+				StripPaths:            false,
+				BackendTimeoutSeconds: 10,
+				BackendMaxConnections: defaultMaxConnections,
+				ProxyBufferSize:       6,
+				ProxyBufferBlocks:     4,
+			}},
+			Config{
+				DefaultBackendTimeoutSeconds: backendTimeout,
+				DefaultProxyBufferSize:       2,
+				DefaultProxyBufferBlocks:     3,
+			},
+		},
+		{
+			"ingress definition resets to max when proxy buffer values exceed max allowed values",
+			createIngressesFixture(ingressHost, ingressSvcName, ingressSvcPort, map[string]string{ingressAllowAnnotation: "", proxyBufferSizeAnnotation: "64", proxyBufferBlocksAnnotation: "12"}),
+			createDefaultServices(),
+			[]IngressEntry{{
+				Namespace:             ingressNamespace,
+				Name:                  ingressName,
+				Host:                  ingressHost,
+				Path:                  ingressPath,
+				ServiceAddress:        serviceIP,
+				ServicePort:           ingressSvcPort,
+				LbScheme:              "",
+				Allow:                 []string{},
+				StripPaths:            false,
+				BackendTimeoutSeconds: 10,
+				BackendMaxConnections: defaultMaxConnections,
+				ProxyBufferSize:       32,
```

Comment:
Is this not the max? Can we set it to a value greater than the max instead? Additionally, it might be worth using the `maxAllowedProxyBuffer*` constants here so that it doesn't break the tests if we raise/drop them.

Reply:
Not sure I follow :( The ingress entry is our expectation. The test is when
 ```
proxyBufferSizeAnnotation: "64",
proxyBufferBlocksAnnotation: "12",
```  is the annotation, we will use 32 and 8 instead